### ISSUE TITLE
Errors are swallowed

### ISSUE THREAD

**Issue by chrismwendt** [sourcegraph/lang-go#2]
https://github.com/sourcegraph/lang-go/blob/70c1fd74e9839ce8251d3dc9f3c640001ccf64b7/src/lang-go.ts#L221-L225

@felixfbecker I'm having some difficulty figuring out why errors are getting swallowed in my rxjs code here. I tried `console.log`ging every Promise and it seems like the Promise rejection gets lost somewhere in:

https://github.com/sourcegraph/lang-go/blob/70c1fd74e9839ce8251d3dc9f3c640001ccf64b7/src/lang-go.ts#L212-L219

But I couldn't figure out why. If you'd like I can share my screen sometime and we can take a look at this.

**Comment by felixfbecker** [comment]
You manually create an Observable here: 

https://github.com/sourcegraph/lang-go/blob/70c1fd74e9839ce8251d3dc9f3c640001ccf64b7/src/lang-go.ts#L183-L193

But in there, I don't see you calling `observer.error()` in any error condition. You only attach `.then()` to call `observer.next()`, but not `.catch()` to call `observer.error()`.
In general I would recommend you to not create Observables manually for this reason, just use `fromPromise()`, which does this correctly.

**Comment by chrismwendt** [comment]
How do I expose `unsubscribe` when using `fromPromise()`? I want to make sure that old connections get disposed of when the langserver address changes.

**Comment by felixfbecker** [comment]
Nvm, I just saw that it actually emits a function that performs the actual request

**Comment by felixfbecker** [comment]
I don't know just from looking at it. It's hard to follow the data flow from reading
